pkg/compiler/types.go

```diff
@@ -83,13 +86,16 @@ var typeInt = &typeDesc{
 	},
 	Gen: func(comp *compiler, t *ast.Type, args []*ast.Type, base prog.IntTypeCommon) prog.Type {
 		size, be := comp.parseIntType(t.Ident)
-		kind, rangeBegin, rangeEnd := prog.IntPlain, uint64(0), uint64(0)
+		kind, rangeBegin, rangeEnd, align := prog.IntPlain, uint64(0), uint64(0), uint64(0)
 		if len(args) > 0 {
 			rangeArg := args[0]
 			kind, rangeBegin, rangeEnd = prog.IntRange, rangeArg.Value, rangeArg.Value
 			if len(rangeArg.Colon) != 0 {
 				rangeEnd = rangeArg.Colon[0].Value
 			}
+			if len(args) > 1 {
+				align = args[1].Value
```

Comment:
This needs some checks. I can think of at least [1:7, 10] that does not make sense.

Reply:
Also align 0 or 1 or negative don't make sense.

Reply:
Okay. I wasn't sure you'd want to enforce this since `align` could be a constant and sometimes be greater than `rangeEnd`, sometimes lower. `[1:7, 10]` would still return `1`.

Reply:
But what's the point of [1:7, 10]? That's just const[1]? Frequently such things mean a typo/bug in descriptions. And since we don't have any real testing for the descriptions, usually such warnings find a number of real bugs.

Reply:
> But what's the point of [1:7, 10]? That's just const[1]?

Not much point. I just wanted to allow for `[1:7, SOME_CONSTANT_FROM_KERNEL]`, but that's probably a corner case and finding bugs in descriptions is more important. I'll add checks!

Reply:
If SOME_CONSTANT_FROM_KERNEL does not vary across archs (or there is a corresponding literal constant for range end), then it's still possible to do, right?
Also may be solvable with defines:
```
define RANGE_END SOME_CONSTANT_FROM_KERNEL * 10
```
Let's first hit such case, relaxing checks is always possible.

Reply:
Okay, that makes sense.

Reply:
I've rebased with an updated patchset. Only subranges of `[-MAX_INTN:MAX_INTN]` for any `intN` are allowed. Is that what you had in mind? I wasn't sure how to treat negative values.